sftp-client.c

```diff
@@ -1193,6 +1222,41 @@ do_download(struct sftp_conn *conn, const char *remote_path,
 
 	attrib_clear(&junk); /* Send empty attributes */
 
+#ifdef IRONCORE
+	if (strlen(local_path) > PATH_MAX - IRON_SECURE_FILE_SUFFIX_LEN) {
```

Comment:
Should you be checking if the file name plus the iron extension exceeds `NAME_MAX`?